src/ui/Base/Component.ts

```diff
@@ -425,6 +425,9 @@ export class ComponentEvents {
         } else if (args && JQueryUtils.isInstanceOfJqueryEvent(args[0])) {
           if (args[1] != undefined) {
             args = [args[1]];
+          } else if (args[0].hasOwnProperty('originalEvent')) {
+            // If the event is a jQuery Event but no args are found, let's use the ones of the originalEvent, if any.
+            args = [args[0].originalEvent];
```

Comment:
That seems like a "small" change, but are you able to list all the differences between what jQuery outputs as an "arg" and what's contained in "originalEvent" ?

Are we sure that it's as simple as "just use it", with no further transformation on the argument object itself ?

They are 100% interchangeable ?

No random side effect could happen because of that ?

Or is it simply a case where you tested in salesforce in a "basic" manner and it seemed okay ?

Reply:
If a jQuery Event has an originalEvent property it means that it have been triggered _directly_ by a browser native event (e.g. click, keydown, etc) (source: Query Cookbook: Solutions & Examples for jQuery Developers from the O'Reilly collection)

Also, you can check with this jsFiddle: https://jsfiddle.net/25esxykp/ that the subsequent event will not carry on the originalEvent.

Therefore,if `args[0].hasOwnProperty('originalEvent')` is true:
1. The event triggered is a jQueryEvent and is args[0] (determined from the usage of `wrapToCallIfEnabled` and how jQuery.on, one, trigger work.
2. The event directly come from a native browser event.
3-a. If the handler is a 'native' event-listener, the first parameter given to it is necessarly an Event object (native), thus, it can deal with the fact that we're giving it to it, even if the handler function didn't take parameter -> *No breaking on native event handler*
3-b. If the handler is a 'jQuery' event-listener, we just didn't give anything in the past to the handler, thus, same as 3-a, if the handler didn't use the args, it can't hurt it to have them. Also, jQuery event-handler function does take an Event -albeit a jQuery one- as its first parameter.


Thus, I'm confident by saying that, yes, there's no random side effect with this method.

Based on what I wrote there, one another solution that could leverage the shouldUseJQueryEvent guard of the `Dom.ts` is to remove altogether line 428 to 432 of this PR. This would leave the args the same without any modifications if there's no additional argument, meaning that it'll be an array with only the jQuery Event object in it, following tightly the specs of jQuery or native event bindings.

Reply:
Alright then !